src/VisionModel/googlenet.jl

```diff
@@ -0,0 +1,52 @@
+using Flux
+
+export googlenet
+
+function conv_block(kernelsize::Tuple{Int64,Int64}, inplanes::Int64, outplanes::Int64; stride::Int64=1, pad::Int64=0)
+  conv_layer = []
+  push!(conv_layer, Conv(kernelsize, inplanes => outplanes, stride = stride, pad = pad))
+  push!(conv_layer, BatchNorm(outplanes, relu))
+  return conv_layer
+end
+
+function inception_block(inplanes, out_1x1, red_3x3, out_3x3, red_5x5, out_5x5, pool_proj)
+  branch1 = conv_block((1,1), inplanes, out_1x1)
+
+  branch2 = Chain(conv_block((1,1), inplanes, red_3x3),
+             conv_block((3,3), red_3x3, out_3x3; pad=1))        
+
+  branch3 = Chain(conv_block((1,1), inplanes, red_5x5),
+             conv_block((5,5), red_5x5, out_5x5; pad=1)) 
+
+  branch4 = Chain(MaxPool((3, 3), stride=1, pad=1),
+             conv_block((1,1), inplanes, pool_proj))
+ 
+  inception_layer = cat(branch1, branch2, branch3, branch4; dims=1)
+
+  return inception_layer
+end
+
+function googlenet()
+  layers = Chain(conv_block((7,7), 3, 64; stride=2, pad=3)...,
+                 MaxPool((3,3), stride=2, pad=1),
+                 conv_block((1,1), 64, 64)...,
+                 conv_block((3,3), 64, 192; pad=1)...,
+                 MaxPool((3,3), stride=2, pad=1),
+                 inception_block(192, 64, 96, 128, 16, 32, 32)...,
```

Comment:
`inception_block` doesn't return an array of layers so no need for `...`.

Reply:
The inception block will return an array of layers. Then the type of inception block will be ``Chain(Any(Tuple64{Array, Array}))``. Do we want this?

Reply:
There is dimension mismatch error too at 2nd branch of inception layer.

Reply:
No we don't want it to return an array/chain of layers. If you look at the diagram in the paper, the output of the inception block is a single operation -- `cat`.